lib/cocina/models/validators/dark_validator.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Cocina
+  module Models
+    module Validators
+      # Validates that shelve and publish file attributes are set to false for dark DRO objects.
+      class DarkValidator
+        def self.validate(clazz, attributes)
+          new(clazz, attributes).validate
+        end
+
+        def initialize(clazz, attributes)
+          @clazz = clazz
+          @attributes = attributes
+        end
+
+        def validate
+          return unless meets_preconditions?
+
+          return if invalid_files.empty?
+
+          raise ValidationError, 'Not all files have dark access and/or are unshelved ' \
+                                 "when object access is dark: #{invalid_filenames}"
+        end
+
+        private
+
+        attr_reader :clazz, :attributes
+
+        def meets_preconditions?
+          dro? && attributes.dig(:access, :view) == 'dark'
+        end
+
+        def dro?
+          (clazz::TYPES & DRO::TYPES).any?
+        rescue NameError
+          false
+        end
```

Comment:
@justinlittman I'm wondering if we can call `clazz.dro?` and lean on the `Checkable` module for this instead.

Reply:
Aha, it's because it's an instance method and we have but a class and some props. Nevermind!